streams/src/test/java/org/apache/kafka/streams/kstream/KStreamBuilderTest.java

```diff
@@ -233,4 +237,46 @@ public void shouldMapStateStoresToCorrectSourceTopics() throws Exception {
         assertEquals(Collections.singletonList("table-topic"), builder.stateStoreNameToSourceTopics().get("table-store"));
         assertEquals(Collections.singletonList(APP_ID + "-KSTREAM-MAP-0000000003-repartition"), builder.stateStoreNameToSourceTopics().get("count"));
     }
+
+    @SuppressWarnings("unchecked")
+    @Test
+    public void shouldAddTimestampExtractorPerSource() throws Exception {
+        TimestampExtractor timestampExtractor = new MockTimestampExtractor();
+        String topicName = "topic";
+        String storeName = "store";
+        KStreamBuilder kStreamBuilder;
+        ProcessorTopology processorTopology;
+
+        kStreamBuilder = new KStreamBuilder();
+        kStreamBuilder.stream(topicName);
+        processorTopology = kStreamBuilder.build(null);
+        assertNull(processorTopology.source(topicName).getTimestampExtractor());
```

Comment:
It's better to split this test into multiple -- here you test if no source specific extractor is set, thus, this should be a test method `sourceExtractorShouldBeNull` (or similar) and the test should end here.

Apply to below tests, too. (split into positive/negative tests and own tests for stream/table -- for stream/table add overload methods same way to test `TopologBuilder.addSource()`